### ISSUE TITLE
Rename should affect references in doc comments

### ISSUE THREAD

**Issue by Dit05** [rust-lang/rust-analyzer#18703]
```rust
/// See [`bar`].
fn foo() {}

/// I am bar.
fn bar() {}
```

Hovering the reference to `bar` in the doc comment shows its description, as expected.

When I rename `bar` (its definition after the fn keyword), the reference to it in `foo`'s doc comment is not changed, so it refers to a non-existent item now.

Would it be possible for renaming to affect doc comments as well?

I am using `rust-analyzer` with the default [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#rust_analyzer) configuration. I'm on `rust-analyzer 1.83.0 (90b35a6 2024-11-26)`.

**Comment by ValouBambou** [comment]
I'm not sure if your comment also includes references in docs examples, so I would like to explicitly note it, but it could be a really cool feature to have.

**Comment by Aidavdw** [comment]
The problem right now is that references to docs live in a very different level from where the LSP functions.
Doc links are invisible to rename, because they exist in a different layer.

The right way to implement this would be by moving the resolving of doc comenments, moving this to the same resolution layer.  That would also yield find-all-references, highlight-related, and call-hierarchy filtering consistency.
For this though, you'd need to move link extraction + resolution down into ide-db. 
That would mean pulling pulldown-cmark into ide-db's Cargo.toml (seems to already a workspace dep though?). 
That's a prety big change, but it seems a reasonable one.

**Comment by ChayimFriedman2** [comment]
No, the right way to implement this is to treat code in doc comments like normal code, which will also solve other issues with highlighting and allow other IDE features. It is more complicated though.

**Comment by Aidavdw** [comment]
That's what i was trying to get at. Point is that you'd need to move the analysis of doctext to find the doclinks.
Ths functionality resides in ide-db, no?
